src/sql/platform/bootstrap/browser/bootstrapService.ts

```diff
@@ -38,26 +39,40 @@ function createUniqueSelector(selector: string): string {
 
 let platform: PlatformRef;
 
-export function bootstrapAngular<T>(service: IInstantiationService, moduleType: IModuleFactory<T>, container: HTMLElement, selectorString: string, params: IBootstrapParams, input?: IEditorInput, callbackSetModule?: (value: NgModuleRef<T>) => void): string {
-	// Create the uniqueSelectorString
-	let uniqueSelectorString = createUniqueSelector(selectorString);
-	let selector = document.createElement(uniqueSelectorString);
-	container.appendChild(selector);
+export function bootstrapAngular<T>(service: IInstantiationService, moduleType: IModuleFactory<T>, container: HTMLElement, selectorString: string, params: IBootstrapParams, input?: IEditorInput, callbackSetModule?: (value: NgModuleRef<T>) => void): Promise<string> {
+	const doBootstrap = ((platform: PlatformRef): string => {
```

Comment:
You shouldn't need to do this whole function wrapping thing. There was no race condition before. The invoke function method runs the callback synchronously.

Reply:
You could even make this whole function an invokable function by just replacing the first param with a service accessor and getting the instantiation service and the environment service directly. Probably a better approach since it means if we need additional services we don't need to keep changing the function definition.